### PR TITLE
Register Dirigible datasources in the Camel context

### PR DESCRIPTION
**Pull request by iliyan-velichkov**
+ add camel dependencies for the products synchronization camel

### CODE REVIEW

File: components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/service/DataSourceLifecycleListener.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.eclipse.dirigible.components.data.sources.service;
+
+import org.eclipse.dirigible.components.data.sources.domain.DataSource;
+
+public interface DataSourceLifecycleListener {
+
+    void onSaved(DataSource dataSource);
```
Comment (delchev) [inline components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/service/DataSourceLifecycleListener.java:7]: let use present tense here: onSave() on onDelete()
usually `on` is associated with `after` when there is only single event used

**Review by delchev — CHANGES_REQUESTED** [review]

File: components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    public boolean isAlreadyInitialized(String dataSourceName) {
+        return DATASOURCES.containsKey(dataSourceName);
+    }
+
+    public javax.sql.DataSource getAlreadyInitializedDataSource(String dataSourceName) {
+        return DATASOURCES.get(dataSourceName);
+    }
+
```
Comment (delchev) [inline components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java:104]: let introduce method for removing already initalized datasource

File: components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+     * Initialize data source.
+     *
+     * @param name the name
+     * @return the {@link javax.sql.DataSource}
+     */
+    public javax.sql.DataSource initialize(DataSource dataSource) {
+        String name = dataSource.getName();
+
+        if (isAlreadyInitialized(name)) {
+            return getAlreadyInitializedDataSource(name);
```
Comment (delchev) [inline components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java:60]: let remove 'Already' from method name

File: components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+        ManagedDataSource managedDataSource = new ManagedDataSource(hds);
+        registerDataSourceBean(name, managedDataSource);
+
+        DATASOURCES.put(name, managedDataSource);
+
+        return managedDataSource;
+    }
+
+    public boolean isAlreadyInitialized(String dataSourceName) {
```
Comment (delchev) [inline components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java:97]: remove `Already` here as well

File: components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/service/DataSourceLifecycleListener.java (hunk 1 above)
Comment (iliyan-velichkov) [reply components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/service/DataSourceLifecycleListener.java]: done

File: components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java (hunk 2 above)
Comment (iliyan-velichkov) [reply components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java:122]: done

File: components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java (hunk 4 above)
Comment (iliyan-velichkov) [reply components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java]: done

File: components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java (hunk 3 above)
Comment (iliyan-velichkov) [reply components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java]: done

**Comment by iliyan-velichkov** [comment]
integration test issue: https://github.com/eclipse/dirigible/issues/3173

**Comment by iliyan-velichkov** [comment]
@ThuF @delchev @vmutafov the pr is for review

**Review by delchev — APPROVED** [review]

**Pull request merged**